File: 2107-find-unique-binary-string/2107-find-unique-binary-string.cpp
class Solution {
public:
map<string,bool>mp;
string ans;
bool func(int indx,int sz,string str){
    if(indx == sz){ 
        if(!mp[str]){
            ans=str;
            return true;
            }
        return false;
    }
    return (func(indx+1,sz,str+'1')||func(indx+1,sz,str+'0'));
}
    string findDifferentBinaryString(vector<string>& nums) {
        string str="";
        int sz=nums[0].size();
        for(auto &it:nums){
            mp[it]=1;
        }
        func(0,sz,str);
        return ans;
    }
};

File: 2107-find-unique-binary-string/2107-find-unique-binary-string.js
/**
 * @param {string[]} nums
 * @return {string}
 */
var findDifferentBinaryString = function(nums) {
     let mp = new Map();
    nums.forEach((val) => {
        mp.set(val, true);
    });
    let sz=nums[0].length;
    let ans="";
    let x= function ( indx, str){
        if(indx == sz){
           if (!mp.has(str)) {
                ans = str;
                return true;
            }
            return false;
        }
        return (x(indx+1,str+'1')||x(indx+1,str+'0'));
    }
    x(0,"");
    return ans;
};